marketplaceItems/tipJar/entityResources/entityClientScripts/tipJar_client.js

```diff
@@ -0,0 +1,401 @@
+        tipJarProps.parentID = _entityID;
+        tipJarProps.position = currentPosition;
+        tipJar = Overlays.addOverlay("model", tipJarProps);
+    }
+    
+    
+    // #endregion
+    // *************************************
+    // END OVERLAY
+    // *************************************
+
+    // *************************************
+    // START USER_DATA
+    // *************************************
+    // #region USER_DATA
+    
+    
+    // Checks to see if the userData is unique
+    function isNewUserDataDifferent(oldUserData, newUserData){
+        oldUserData = typeof oldUserData === "string" ? oldUserData : JSON.stringify(oldUserData);
+        newUserData = typeof newUserData === "string" ? newUserData : JSON.stringify(newUserData);
+        if (oldUserData === newUserData){
+            return false;
+        }
+
+        return true;
+    }
+
+
+    // Get the latest userData before initiating payment
+    var userData = {};
+    var previousUserData = {};
+    var destinationName = null;
+    var hfcAmount = 0;
+    var message = "";
+    function getLatestUserData(){
+        var newUserData = Entities.getEntityProperties(_entityID, 'userData').userData;
+        if (isNewUserDataDifferent(previousUserData, newUserData)){
+            previousUserData = "name" in userData ? userData : newUserData;
+            userData = JSON.parse(newUserData);
+
+            // In case someone puts something strange in for the number
+            if (typeof userData.hfcAmount !== "number") {
+                userData.hfcAmount = 1;
+            }
+
+            // hfc shows on inventory that it gets rounded, so go ahead and round the amount to clear up any confusion
+            hfcAmount = Math.round(userData.hfcAmount);
+            destinationName = userData.destinationName;
+            message = userData.message;
+        }
+    }
+
+    
+    // #endregion
+    // *************************************
+    // END USER_DATA
+    // *************************************
+
+    // *************************************
+    // START ENTITY DEFINITION
+    // *************************************
+    // #region ENTITY DEFINITION
+    
+
+    // Grab the entityID on preload
+    var COIN_SOUND_URL = Script.resolvePath('../resources/sounds/CoinsDrop_BW.6645_1497719.4.R.wav');
+    var CLICK_SOUND_URL = Script.resolvePath('../resources/sounds/beep.mp3');
+    var _entityID = null;
+    var coinSound = null;
+    var clickSound = null;
+    function preload(entityID) {
+        _entityID = entityID;
+        coinSound = SoundCache.getSound(COIN_SOUND_URL);
+        clickSound = SoundCache.getSound(CLICK_SOUND_URL);
+        createTipJarOverlay();
+        getLatestUserData();
+    }
+
+
+    // Main user interaction handler
+    var clickInjector = null;
+    var WAIT_TO_SEND_TIP_TIMEOUT_MS = 500;
+    function userClicked() {
+        clickInjector = playAudio(clickSound);
```

Comment:
this sound seems very quiet to me when using it - am i the only one who thinks that?

Reply:
No I thought so as well, I got the file from Ryan, but the db is pretty low.  Going to bump the actual file 6-12db.